crates/hc_seed_bundle/src/unlocked_seed_bundle.rs

```diff
@@ -20,94 +20,120 @@ const KDF_CONTEXT: &[u8; 8] = b"SeedBndl";
 /// [UnlockedSeedBundle::lock] and supply the desired SeedCiphers.
 #[derive(Clone)]
 pub struct UnlockedSeedBundle {
-    seed: sodoken::BufReadSized<32>,
-    sign_pub_key: sodoken::BufReadSized<{ sodoken::sign::PUBLICKEYBYTES }>,
-    sign_sec_key: sodoken::BufReadSized<{ sodoken::sign::SECRETKEYBYTES }>,
+    seed: SharedSizedLockedArray<32>,
+    sign_pub_key: Arc<[u8; sodoken::sign::PUBLICKEYBYTES]>,
+    sign_sec_key: SharedSizedLockedArray<{ sodoken::sign::SECRETKEYBYTES }>,
     app_data: Arc<[u8]>,
 }
 
 impl UnlockedSeedBundle {
     /// Private core constructor
     pub(crate) async fn priv_from_seed(
-        seed: sodoken::BufReadSized<32>,
-    ) -> SodokenResult<Self> {
+        seed: sodoken::SizedLockedArray<32>,
+    ) -> Result<Self, OneErr> {
+        let seed = Arc::new(Mutex::new(seed));
+
         // generate the deterministic signature keypair represented by this seed
-        let pk = sodoken::BufWriteSized::new_no_lock();
-        let sk = sodoken::BufWriteSized::new_mem_locked()?;
-        sodoken::sign::seed_keypair(pk.clone(), sk.clone(), seed.clone())
-            .await?;
+        let (pk, sk) = tokio::task::spawn_blocking({
+            let seed = seed.clone();
+            move || -> Result<_, OneErr> {
+                let mut pk = [0; sodoken::sign::PUBLICKEYBYTES];
+                let mut sk = sodoken::SizedLockedArray::<
+                    { sodoken::sign::SECRETKEYBYTES },
+                >::new()?;
+                sodoken::sign::seed_keypair(
```

Comment:
The `blocking_argon2id` call is the only one that should need to be wrapped in a spawn_blocking call.